bot/exts/evergreen/profile_pic_modification/pfp_modify.py

```diff
@@ -0,0 +1,307 @@
+    """
+    Runs the given synchronus function `func` in an executor.
+
+    This is useful for running slow, blocking code within async
+    functions, so that they don't block the bot.
+    """
+    log.trace(f"Running {func.__name__} in an executor.")
+    loop = asyncio.get_event_loop()
+    return await loop.run_in_executor(_EXECUTOR, func, *args)
+
+
+class PfpModify(commands.Cog):
+    """Various commands for users to change their own profile picture."""
+
+    def __init__(self, bot: commands.Bot) -> None:
+        self.bot = bot
+
+    async def _fetch_member(self, member_id: int) -> t.Optional[discord.Member]:
+        """
+        Fetches a member and handles errors.
+
+        This helper funciton is required as the member cache doesn't always have the most up to date
+        profile picture. THis can lead to errors if the image is delted from the Discord CDN.
+        """
+        try:
+            member = await self.bot.get_guild(Client.guild).fetch_member(member_id)
+        except discord.errors.NotFound:
+            log.debug(f"Member {member_id} left the guild before we could get their pfp.")
+            return None
+        except discord.HTTPException:
+            log.exception(f"Exception while trying to retrieve member {member_id} from Discord.")
+            return None
+
+        return member
+
+    @commands.group()
+    async def pfp_modify(self, ctx: commands.Context) -> None:
+        """Groups all of the pfp modifying commands to allow a single concurrency limit."""
+        if not ctx.invoked_subcommand:
+            await invoke_help_command(ctx)
+
+    @pfp_modify.command(name="8bitify", root_aliases=("8bitify",))
+    async def eightbit_command(self, ctx: commands.Context) -> None:
+        """Pixelates your avatar and changes the palette to an 8bit one."""
+        async with ctx.typing():
+            member = await self._fetch_member(ctx.author.id)
+            if not member:
+                ctx.send(f"{Emojis.cross_mark} Could not get member info.")
+                return
+
+            image_bytes = await member.avatar_url.read()
+            file_name = FILENAME_STRING.format(
+                effect="eightbit_avatar",
+                author=member.display_name
+            )
+
+            file = await in_executor(
+                PfpEffects.apply_effect,
+                image_bytes,
+                PfpEffects.eight_bitify_effect,
+                file_name
+            )
+
+            embed = discord.Embed(
+                title="Your 8-bit avatar",
+                description="Here is your avatar. I think it looks all cool and 'retro'."
+            )
+
+            embed.set_image(url=f"attachment://{file_name}")
+            embed.set_footer(text=f"Made by {member.display_name}.", icon_url=member.avatar_url)
+
+        await ctx.send(embed=embed, file=file)
+
+    @pfp_modify.command(pass_context=True, aliases=["easterify"], root_aliases=("easterify", "avatareasterify"))
```

Comment:
```suggestion
    @pfp_modify.command(aliases=["easterify"], root_aliases=("easterify", "avatareasterify"))
```
`pass_context` is no longer a valid kwarg, as it's passed automatically,